WalletWasabi/Helpers/PowerSaving/MacOsInhibitorTask.cs

```diff
@@ -0,0 +1,38 @@
+using System.Diagnostics;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+
+namespace WalletWasabi.Helpers.PowerSaving;
+
+/// <summary>
+/// Inhibitor based on <c>caffeinate</c> command.
+/// </summary>
+public class MacOsInhibitorTask : BaseInhibitorTask
+{
+	private MacOsInhibitorTask(TimeSpan period, string reason, ProcessAsync process)
+		: base(period, reason, process)
+	{
+	}
+
+	public static MacOsInhibitorTask Create(TimeSpan basePeriod, string reason)
+	{
+		string innerCommand = $$"""
+			\$(caffeinate -i &) ;
+			caffeinatePid=\$!;
+			trap \"kill -9 \$caffeinatePid\" 0 SIGINT SIGTERM;
+			wait {{Environment.ProcessId}};
+			""".ReplaceLineEndings(replacementText: " ");
```

Comment:
![giphy (3)](https://user-images.githubusercontent.com/9844978/206227523-64624810-4be6-4e76-b431-c437117aefbc.gif)

Are you sure?

Reply:
Can you add a comment on what is happening here?